### PR TITLE
fix(ticket rule): action status + solution template

### PR DESCRIPTION
**Pull request by Rom1-B**
In business rules for tickets, actions to change status and add solution template can be used in same rules but are not compatible.
The status change is done before the solution is added, and therefore the solution fails, because the ticket is already resolved.
In this case (when the status is forced) adding the template should not compute the status again.

| Q             | A
| ------------- | ---
| Bug fix?      | yes
| New feature?  | no
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | yes
| Fixed tickets | !25009

### CODE REVIEW

File: src/ITILSolution.php [hunk 1]
```diff
@@ -220,6 +220,9 @@ public function prepareInputForAdd($input)
             $input = array_replace($template_fields, $input);
         }
 
+        if (isset($input['_do_not_compute_status'])) {
+            return $input;
```
Comment (cedric-anne) [inline src/ITILSolution.php:224]: There is a piece of code that is not related to status on line 264. Instead of returning here it would be preferable to encapsulate the whole status update logic in a `if (!this->item->isStatusComputationBlocked($input)) {}` block.

**Review by cedric-anne — APPROVED** [review]

**Review by trasher — APPROVED** [review]

**Review by cconard96 — APPROVED** [review]
